pkg/sync/sync_context.go

```diff
@@ -400,6 +410,15 @@ func (sc *syncContext) Sync() {
 
 	sc.log.WithValues("tasks", tasks).V(1).Info("Wet-run")
 	runState := sc.runTasks(tasks, false)
+
+	if sc.syncWaveHook != nil && runState != failed {
```

Comment:
Here if `runState` is failed, `SyncWaveHook` is not called. I wonder if there is a use case that `SyncWaveHook `needs the flexibility to do different things based on the `runState`? Do we want the input parameter of SyncWaveHook also includes runState?